store/src/main/java/org/apache/rocketmq/store/timer/TimerMessageStore.java

```diff
@@ -599,7 +599,12 @@ public void addMetric(MessageExt msg, int value) {
             if (null == msg || null == msg.getProperty(MessageConst.PROPERTY_REAL_TOPIC)) {
                 return;
             }
-            timerMetrics.addAndGet(msg.getProperty(MessageConst.PROPERTY_REAL_TOPIC), value);
+            if (msg.getProperty(TIMER_ENQUEUE_MS) != null
```

Comment:
NumberUtils.toLong() might be more clear

Reply:
done

Reply:
no need of `msg.getProperty(TIMER_ENQUEUE_MS) != null` if using `NumberUtils.toLong`, you can check this out

Reply:
Under normal situation, this indicator is still null when exiting the queue, so this judgment is necessary.
When leaving the queue, the 'enququ_time' attribute in the message has not been retrieved from the timerRequest yet. When leaving the queue, this attribute will not be null unless it is determined that it has never entered the time wheel.